test/common/buffer/buffer_fuzz_test.cc

```diff
@@ -274,7 +274,7 @@ uint32_t bufferAction(Context& ctxt, char insert_value, uint32_t max_alloc, Buff
     Buffer::RawSlice slices[1];
     const uint64_t slices_used = target_buffer.getRawSlices(slices, 1);
     if (linearize_size > 0) {
-      FUZZ_ASSERT(slices_used == 1);
+      FUZZ_ASSERT(slices_used >= 1);
```

Comment:
Some context on this change: the buffer API specifies that `linearize(n)` will linearize at most the first `n` bytes of the buffer. The libevent buffer implementation happens to linearize the entire buffer; but the new buffer implementation stops at the next slice boundary after `n` bytes. The next line after this verifies that the first `n` bytes have indeed been linearized.

Reply:
Just to sanity check, have we verfied that the in-envoy code sites either linearize(full buffer) or lineraize(n)/write(n) i.e. this change will end up being a no-op?

Reply:
Anyway, most of the call sites in Envoy use a third form instead: "linearize one field." But this is also safe with both the old and new implementations (and with any future implementation that satisfies the semantics defined in `buffer.h`) :
```
void* mem = buffer.linearize(n);
// read/interpret the n-byte field value
```
[Here](https://gist.github.com/brian-pane/07aa0a09b33b62dccad065aaf825d9a2) are all the `linearize` calls in Envoy. From that list, every call site uses one of those three safe patterns:
* All the call sites in mongo_proxy safely use the "linearize one field" pattern.
* dubbo_proxy provides a buffer wrapper class whose `linearize` method that maintains an internal offset within the buffer. That method itself looks correct, as long as its callers only rely on the documented `linearize` semantics.
* The http, mysql, and thrift filters safely use the "linearize one field" pattern.
* `source/extensions/transport_sockets/tls/ssl_socket.cc` uses the "linearize(n)/write(n)" pattern.
* `source/extensions/transport_sockets/alts/tsi_handshaker.cc` uses the "linearize(n)/write(n)" pattern. (The write is encapsulated in a gRPC library function, but the Envoy code passes the right value of "n" to it.)
* `source/extensions/transport_sockets/alts/tsi_frame_protector.cc` uses "linearize(buffer.length())"

So all the call sites in Envoy do the right thing. That's good, because even independent of this PR, the old implementation can't be assumed to linearize the entire buffer either. The underlying libevent function it uses, `evbuffer_pullup`, is only documented as linearizing the first `n` bytes. It currently happens to linearize the whole buffer in some test cases where `n` is close to the buffer length, but that's not guaranteed in the general case or in future releases of libevent.